### DISCUSSION TITLE
faster way to check if is null

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#5645] [Language Ideas] [answered]
in
```if(A==null)``` or ```if(A is null)```
there is a lot of letters
we can reduce it to 30% by using ```if(?A)``` or ```if(A?)``` or ```A? {..do something} : {...do something}```
where as ```?``` checks for null or false

**Comment by CyrusNajmabadi** [comment]
> there is a lot of letters

This is not a problem.

> we can reduce it to 30% by using if(?A)

The existing options are clear, idiomatic, and used pervasively in the ecosystem.  This is not a problem we would desire to solve.  We would just end up with one more way of doing the same thing.

**Comment by theunrepentantgeek** [comment, marked as answer]
Code is read many (many!) more times than it is written.

Cognitive science research into literacy shows (indirectly) that `if (A is null)` will be read more reliably - and more quickly - than the suggested form. (Using `is null` may also be faster than `if (A == null)`, but that's dependent on whether the reader has internalised punctuation like `==` as a letter form or not.)

It would literally be a case where the number of keystrokes would be reduced at the cost of decreasing readability.

**Reply by Xyncgas** [reply]
Sometimes I just lose a lot sense of meaning in life when I can't just compare two things and keep working on other things
